### PR TITLE
Vert.x HTTP Security - don't use deprecated classes and methods and remove unused classes

### PR DESCRIPTION
**Pull request by michalvavrik**
Changes:

- removed `io.quarkus.vertx.http.runtime.security.SupplierImpl` as it has no function and we don't use it
- `io.quarkus.vertx.http.runtime.security.QuarkusHttpUser#isAuthorized(java.lang.String, io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Boolean>>)` method is deprecated in Vert.x so I deprecated it as well
- removed `io.quarkus.vertx.http.runtime.security.QuarkusHttpSecurityContext` as it has no function and we don't use it
- marked `io.quarkus.vertx.http.runtime.security.HttpAuthenticationMechanism#getCredentialTransport()` as "for removal" as it has been deprecated since Quarkus 2.8 https://github.com/quarkusio/quarkus/pull/22483
- converted `io.quarkus.vertx.http.runtime.security.AbstractPathMatchingHttpSecurityPolicy.HttpMatcher` to record, it's package-private class so it shouldn't be relevant (or important at all, I just think it's less verbose)
- marked `io.quarkus.vertx.http.deployment.HttpSecurityPolicyBuildItem` for removal, it has been deprecated since 3.6 
and this build item is quite useless now, integrating extensions can define a CDI bean like any other Quarkus user
- replaced usage of `io.quarkus.arc.deployment.AnnotationsTransformerBuildItem#AnnotationsTransformerBuildItem(io.quarkus.arc.processor.AnnotationsTransformer)` because it is marked for removal

### CODE REVIEW

**Review by sberyozkin — APPROVED** [review]

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java [hunk 1]
```diff
@@ -50,7 +50,7 @@
  * Class that is responsible for running the HTTP based authentication
  */
 @Singleton
-public class HttpAuthenticator {
+public final class HttpAuthenticator {
```
Comment (cescoffier) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java:53]: Any reason? That would avoid users to reuse parts of the logic.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthorizer.java [hunk 2]
```diff
@@ -20,7 +20,7 @@
  * Class that is responsible for running the HTTP based permission checks
  */
 @Singleton
-public class HttpAuthorizer extends AbstractHttpAuthorizer {
+public final class HttpAuthorizer extends AbstractHttpAuthorizer {
```
Comment (cescoffier) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthorizer.java:23]: I would say the same as for the HttpAuthenticator - but because there is an abstract class, it should be fine right? The user can still extends the abstract class.

**Review by cescoffier — CHANGES_REQUESTED** [review]
LGTM

I added 2 comments regarding classes now marked as final (which could be a breaking change). 

I like the extended the @Deprecated with the `since` and `forRemoval` attribute.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java (hunk 1 above)
Comment (michalvavrik) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java:53]: > Any reason? 

TL;DR; when it is not final, users can start reusing it, but we change it. Long version (you don't need to read it):

- It's runtime class that uses constructor for CDI injection.
- We changed this class many times during last few releases
- I thought setting `final` gives clear message that it should not be extended
- recently we had user that reported change of class in OIDC in runtime module, before we had user that reported that I removed some method (I think it was on this very class)

> That would avoid users to reuse parts of the logic.

It's fine if we expect it, but if with every change we need to consider users that may have used this class, it means we cannot do whatever in please in runtime module.

Whatever I add, I try to make clear line (usually by visibility or by `sealed`) what I expect users to use and what I don't expect, so that users are not surprised when it is changed.

But I definitely don't need to do changes like this to existing classes, so if you want, I'll just drop it.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthorizer.java (hunk 2 above)
Comment (michalvavrik) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthorizer.java:23]: Same answer as above. I don't need to do it on existing class if you disagree with arguments.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java (hunk 1 above)
Comment (cescoffier) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java:53]: Ok, so it would need to be added in the migration guide (usages of these classes were not expected, and these classes should be considered only for internal use)

**Review by cescoffier — APPROVED** [review]

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java (hunk 1 above)
Comment (michalvavrik) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java:53]: Agreed, but I cannot adjust migration guide myself. @sberyozkin please either add note to the migration guide or I could drop this change. I make these sneaky visibility / `final` changes quite often, so you probably didn't realize I changed it :-)
